src/components/PressableWithSecondaryInteraction/index.native.js

```diff
@@ -0,0 +1,33 @@
+import _ from 'underscore';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Pressable} from 'react-native';
+
+const propTypes = {
+    // The function that should be called when this pressable is LongPressed or right-clicked.
+    onSecondaryInteraction: PropTypes.func.isRequired,
+
+    // The children which should be contained in this wrapper component.
+    children: PropTypes.node.isRequired,
+};
+
+/**
+ * This is a special Pressable that calls onSecondaryInteraction when LongPressed.
+ * @returns {React.Component}
+ */
+const PressableWithSecondaryInteraction = ({onSecondaryInteraction, children, ...props}) => (
+    <Pressable
+        onLongPress={(e) => {
+            e.preventDefault();
+            onSecondaryInteraction(e);
+        }}
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...(_.omit(props, 'onLongPress'))}
+    >
+        {children}
+    </Pressable>
+);
+
+PressableWithSecondaryInteraction.propTypes = propTypes;
```

Comment:
Add a `displayName` to this?

Reply:
Doesn't look like that's necessary for a wrapper component:

![image](https://user-images.githubusercontent.com/47436092/105563988-17156680-5cd5-11eb-9e7b-8eb7cfa19f7c.png)

Reply:
Interesting. It's a stateless component and we usually add the `displayName`.

Reply:
Why do we not need it in this case?

Reply:
The plot thickens! Adding `dispalyName` on stateless components is not something we've been consistent about in this codebase so far. Some examples of stateless components without `displayName` are:

1) `src/components/Avatar.js`
2) `src/pages/home/report/ReportActionItem.js`
3) `src/components/TextInputWithFocusStyles.js`
4) `src/components/ThumbnailImage.js`

There are probably more examples because I didn't go look through every file in the repo. Anyways, I'm happy to add the display name here because it doesn't do any harm, but this inconsistency is something we should make a decision on and address at some point.

Reply:
Interesting, not 100% sure but I _think_ I figured out the story here. If a stateless component uses a regular function (not an arrow function), then it will use the function's name as the display name. If it uses an arrow function, you need to provide a display name. However, all our code (native and web/desktop I believe), goes through a babel pre-processor which converts all ES6 arrow functions into regular named functions. So the display name isn't necessary anymore.

Reply:
That also explains why it was never necessary for class components.

Reply:
>Adding dispalyName on stateless components is not something we've been consistent about in this codebase so far

Consistency will never be perfect and always will be us doing our best to make things consistent. Let's do what we can and if we spot inconsistency we can address it in reviews or create issues to clean things up.

>goes through a babel pre-processor which converts all ES6 arrow functions into regular named functions

Yes, but what we see on dev is not always what we see on production.

**Dev**
<img width="793" alt="Screen Shot 2021-01-25 at 9 19 29 AM" src="https://user-images.githubusercontent.com/32969087/105754764-8ade8b80-5eee-11eb-829c-99af8bfe0ac5.png">

**Production**
<img width="1152" alt="Screen Shot 2021-01-25 at 9 17 30 AM" src="https://user-images.githubusercontent.com/32969087/105754780-90d46c80-5eee-11eb-9d8b-6fb68ff97d20.png">

Reply:
Maybe we should also add something to `STYLE.md` about why the `displayName` is important now that we know the reason.

Reply:
Cool. Added the display name 👍